fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java

```diff
@@ -563,11 +569,31 @@ public void replaceProperties(final IdentifierConverter<Resource, FedoraResource
             exceptions.append(e.getMessage());
         }
 
+        removeEmptyFragments();
+
         if (exceptions.length() > 0) {
             throw new MalformedRdfException(exceptions.toString());
         }
     }
 
+    private void removeEmptyFragments() {
+        try {
+            if (node.hasNode("#")) {
```

Comment:
Oh, that's the use of `getNode`, which throws `PathNotFoundException` when there is no node. That's a crap API. You could do `getNodes("#")` but that's not really any better. Oh, well.